pipeline/coordinator.go

```diff
@@ -377,26 +378,26 @@ func checkClipResolution(p UploadJobPayload, inputVideoProbe *video.InputVideo,
 	}
 }
 
-func ShouldGenerateMP4(sourceURL, mp4TargetUrl *url.URL, fragMp4TargetUrl *url.URL, mp4OnlyShort bool, durationSecs float64) bool {
+func ShouldGenerateMP4(sourceURL, mp4TargetUrl *url.URL, fragMp4TargetUrl *url.URL, mp4OnlyShort bool, durationSecs float64) (bool, string) {
 	// Skip mp4 generation if we weren't able to determine the duration of the input file for any reason
 	if durationSecs == 0.0 {
-		return false
+		return false, "duration is missing or zero"
 	}
 	// We're currently memory-bound for generating MP4s above a certain file size
 	// This has been hitting us for long recordings, so do a crude "is it longer than 12 hours?" check and skip the MP4 if it is
 	if clients.IsHLSInput(sourceURL) && durationSecs > maxRecordingMP4Duration.Seconds() {
-		return false
+		return false, "recording duration is too long"
 	}
 
 	if mp4TargetUrl != nil && (!mp4OnlyShort || durationSecs <= maxMP4OutDuration.Seconds()) {
-		return true
+		return true, ""
 	}
 
-	if fragMp4TargetUrl != nil {
-		return true
+	if fragMp4TargetUrl == nil {
+		return false, "missing MP4 target URL"
 	}
 
-	return false
+	return true, ""
```

Comment:
Flipped the logic to make it more of a standard Golang flow of error checking and then success